src/main/java/net/greghaines/jesque/worker/Watchdog.java

```diff
@@ -74,37 +71,52 @@ private void activate() {
 
     private void requeueLostJobs(final String serverName, final String isDebug) {
         // On watchdog activation check whether server contains unhandled in-flight jobs and re-queuing them
-        try {
-            jedis.evalsha(this.watchdogScriptHash.get(), WATCHDOG_SCRIPT_KEYS_NUMBER, serverName, REQUEUE_JOBS, isDebug);
-        } catch (RuntimeException e) {
-            LOG.error("Failed to run " + REQUEUE_JOBS + " job " + WATCHDOG_LUA + " script", e);
-            recoverFromException(e);
-            requeueLostJobs(serverName, isDebug);
+
+        int requeueJobsRetryNum = 0;
+
+
+        while (requeueJobsRetryNum <= REQUEUE_JOBS_RETRY_NUM) {
+            try {
+                doWorkInPool(this.jedisPool, (PoolUtils.PoolWork<Jedis, Void>) jedis -> {
+                    jedis.evalsha(watchdogScriptHash.get(), WATCHDOG_SCRIPT_KEYS_NUMBER, serverName, REQUEUE_JOBS, isDebug);
+                    return null;
+                });
+                break;
```

Comment:
This is a bad practice.. using break to get out of a loop.
This makes code less readable as you have multiple exit point from the loop to look at...